File: 3174.py
horas_bonecos = 0
horas_arquitetos = 0
horas_musicos = 0
horas_desenhistas = 0

for i in range(int(input())):
  elfo, grupo, horas = input().split()
  horas = int(horas)

  if grupo == "bonecos":
    horas_bonecos += horas
  elif grupo == "arquitetos":
    horas_arquitetos += horas
  elif grupo == "musicos":
    horas_musicos += horas
  elif grupo == "desenhistas":
    horas_desenhistas += horas

qtd_presentes = 0
qtd_presentes += horas_bonecos // 8
qtd_presentes += horas_arquitetos // 4
qtd_presentes += horas_musicos // 6
qtd_presentes += horas_desenhistas // 12

print(qtd_presentes)

File: 3174.c
#include <stdio.h>
#include <string.h>

int main() {
  int n, h, b, a, m, d;
  char g[20];

  scanf("%d", &n);

  b = 0;
  a = 0;
  m = 0;
  d = 0;

  for (int i = 0; i < n; i++) {
    scanf("%*s %s %d", g, &h);
    
    if (strcmp(g, "bonecos") == 0) {
      b = b + h;
    } else if (strcmp(g, "arquitetos") == 0) {
      a = a + h;
    } else if (strcmp(g, "musicos") == 0) {
      m = m + h;
    } else if (strcmp(g, "desenhistas") == 0) {
      d = d + h;
    }
  }

  int p = 0;
  p = p + b / 8;
  p = p + a / 4;
  p = p + m / 6;
  p = p + d / 12;

  printf("%d\n", p);

  return 0;
}